File: 프로그래머스/lv3/43162. 네트워크/네트워크.py
def solution(n, computers):
    answer = 0
    visited = [[False] * n for _ in range(n)]

    def dfs(x, y):
        visited[x][y] = True
        for i in range(len(computers[y])):
            if computers[y][i] == 1 and not visited[y][i]:
                dfs(y, i)

    for i in range(n):
        for j in range(n):
            if not visited[i][j] and computers[i][j] == 1:
                answer += 1
                dfs(i, j)

    return answer

File: 프로그래머스/3/43162. 네트워크/네트워크.java
class Solution {
    boolean[][] visited;
    public int solution(int n, int[][] computers) {
        int answer = 0;
        visited = new boolean[n][n];
        
        for (int i = 0 ; i < n; i ++) {
            for (int k = 0; k < n ; k ++) {
                if (computers[i][k] == 1 && visited[i][k] == false) {
                    answer ++;
                    dfs(i, k ,computers, n);
                }
            }
        }
        
        return answer;
    }
    public void dfs(int x, int y, int[][] computers, int n) {
        visited[x][y] = true;
        
        for (int i = 0 ; i < n ; i++) {
            if (computers[y][i] == 1 && !visited[y][i]) {
                System.out.println(y + "," + i);
                dfs(y, i, computers, n);
            }
        }
        
    }
}